python/pyspark/sql/connect/expressions.py

```diff
@@ -89,7 +90,16 @@ class Expression:
     """
 
     def __init__(self) -> None:
-        pass
+        origin = current_origin()
+        fragment = origin.fragment
```

Comment:
For https://github.com/apache/spark/pull/46789#issuecomment-2175453126:

Ah, okay. Now I remember why we came this way. The problem is that there's no way to get the caller API name within, say, `Expression.__init__()` so we should either:
1. Manually pass the function name or
2. Wrap individual API method (but do not use threadlocal).
    The initial implementation of Spark Connect was like this https://github.com/apache/spark/pull/45377, then I realised that this is actually error-pone, and flaky (e.g., it needed a followup: https://github.com/apache/spark/pull/46053). I pointed it out here: https://github.com/apache/spark/pull/45377#issuecomment-2056191793

My take was that either 1. or 2. have the same flakiness, and replies on some manual string name so we went this way (in addition to match with Scala implementation).